### DISCUSSION TITLE
How to get the target branch name using pull_request_review event on github actions

### DISCUSSION THREAD

**Discussion by so-amuk** [community/community#27154] [Actions]
<p>I have multiple jobs in github actions. I want to run jobs based on specific branches only on <strong>pull_request_review</strong> event.  I tried using <code>github.ref</code>, <code>github.head_ref</code>,  <code>github.base_ref</code> but none of them worked as they only work on <strong>pull_request</strong> or <strong>push</strong> event. As it is described below:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d4dfafe56c8d8c2980556f0a95c94f8f1ad2a538.png" data-download-href="/uploads/short-url/unatd0fhBfSoF0iwehHqEk2Rswg.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/39334174/181102084-303fe3df-6656-4c07-b391-cc0502fdea89.png" alt="image" data-base62-sha1="unatd0fhBfSoF0iwehHqEk2Rswg" width="690" height="287" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d4dfafe56c8d8c2980556f0a95c94f8f1ad2a538_2_690x287.png, https://aws1.discourse-cdn.com/github/original/2X/d/d4dfafe56c8d8c2980556f0a95c94f8f1ad2a538.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/d4dfafe56c8d8c2980556f0a95c94f8f1ad2a538.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d4dfafe56c8d8c2980556f0a95c94f8f1ad2a538_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">807×336 41 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So my question is how to get target branch name?</p>
<p>I created PR from branch “latest” to “master”</p>
<p>I tried following, but does not work for me. It skips the first job <strong>Job1</strong></p>
<pre><code class="lang-auto">name: Github actions
on: 
    pull_request_review:
      types: [submitted]
jobs:
    Job1:
        name: Job1
        runs-on: ubuntu-latest
        if: github.ref == 'refs/heads/master'
        steps:
            -   name: trigger
                run: echo "Hii"
    Job2:
        name: Job2
        runs-on: ubuntu-latest
        if: github.ref != 'refs/heads/master'
        steps:
            -   name: trigger
                run: echo "Hello"
</code></pre>
<p>Then I tried different approach  and that also doesn’t work:</p>
<p>job1.yaml</p>
<pre><code class="lang-auto">name: Github actions
on: 
    pull_request_review:
      types: [submitted]
    branches:
       - 'master'
jobs:
    Job1:
        name: Job1
        runs-on: ubuntu-latest
        steps:
            -   name: trigger
                run: echo "Hii"
</code></pre>
<p>job2.yaml</p>
<pre><code class="lang-auto">name: Github actions
on: 
    pull_request_review:
      types: [submitted]
    branches:
       - '*'
       - '!master'
jobs:
    Job2:
        name: Job2
        runs-on: ubuntu-latest
        steps:
            -   name: trigger
                run: echo "Hii"
</code></pre>
<p>So after trying these steps, none of them worked. Now I am looking for different approach of how to get target branch name and then use conditions based on branches and then run the jobs. If I can use the same approach then how to achieve it? If anyone has suggestion how to achieve this using same or different approach then please answer. Thanks in advance!</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/so-amuk">@so-amuk</a>,</p>
<p>For pull_request_review event, please use ${{ github.event.pull_request.base.ref }} to get the target branch name.</p>
<p>The value comes from Github Context, for different event, the value name and location could be different. Please use below code to export the context:</p>
<pre><code class="lang-auto">      - name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
</code></pre>
<p>Follow the structure of the context, you can find the target branch name.<br>
So your code can be fixed as below:</p>
<pre><code class="lang-auto">name: Github actions
on: 
    pull_request_review:
      types: [submitted]
jobs:
  Job1:
    name: Job1
    runs-on: ubuntu-latest
    if: github.event.pull_request.base.ref == 'master'   # check the target branch if it's master
    steps:
        - name: trigger
          run: echo "Hii"
  Job2:
    name: Job2
    runs-on: ubuntu-latest
    if: github.event.pull_request.base.ref != 'master' # check the target branch if it's master
    steps:
        - name: trigger
          run: echo "Hello"
</code></pre>
<p>Please check my workflow for your reference.<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/weide-zhou/Ticket2/runs/852681705?check_suite_focus=true#step:2:549" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181102086-324fe50e-d65e-42a6-ae5d-0fe21648403b.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/weide-zhou/Ticket2/runs/852681705?check_suite_focus=true#step:2:549" target="_blank" rel="nofollow noopener">weide-zhou/Ticket2</a></h3>

<p>Contribute to weide-zhou/Ticket2 development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Thanks</p>

**Comment by so-amuk** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> Thanks for your valuable answer. It worked for me.</p>

**Comment by tb102122** [comment]
<p>Thanks for the hint.<br>
Below the revised version for Windows.</p>
<pre><code class="lang-auto">      - name: Check for Changes
        id: git_diff
        run: |
          if ( $env:github_base_ref ) {
            # Pull Request
            git fetch origin "${{ github.base_ref }}" --depth=1
            $diff_result = (git diff --name-only "origin/${{ github.base_ref }}" ${{ github.sha }} )
            echo "Diff between origin/${{ github.base_ref }} and ${{ github.sha }}"
          }else {
            # Push
            git fetch origin ${{ github.event.before }} --depth=1
            $diff_result = (git diff --name-only ${{ github.event.before }} ${{ github.sha }} )
            echo "Diff between ${{ github.event.before }} and ${{ github.sha }}"
          }
          # echo $diff_result
          $search_result = [bool] ($diff_result | Select-String -Pattern '(CHANGELOG.md)|(.ps1)' -Quiet)
          # echo $search_result
          echo "::set-output name=diff::$search_result"
</code></pre>

**Comment by dememax** [comment]
In our case, it's even not base, but `github.event.pull_request.head.ref`.

As far as I can see, it's not in the official documentation yet?
